lib/Backends/OpenCL/kernels_fwd_conv.cl

```diff
@@ -319,4 +335,221 @@ conv_forward_mem(__global void *mem, unsigned im_in_offset, unsigned wg_offset,
     }
   }
 }
+
+
+__kernel
+__attribute__((reqd_work_group_size(workgroup_size_0, workgroup_size_1, 1)))
+__attribute__((vec_type_hint(Dtype4)))
+void
+conv_forward_mem_i8(__global void *mem, unsigned im_in_offset, unsigned wg_offset,
+                 unsigned bias_offset, unsigned im_out_offset,
+                 int Aoffset, float Ascale, int Boffset, float Bscale,
+                 int Coffset, float Cscale, int Doffset, float Dscale) {
+  __global const Dtype *im_in = &mem[im_in_offset];
+  __global const Dtype *wg = &mem[wg_offset];
+  __global const Dtype *bias = &mem[bias_offset];
+  __global Dtype *im_out = &mem[im_out_offset];
+  // Thread identifiers.
+  // Local row ID (max: RTSM=TSM/WPTM).
+  const int_tp tidn = get_local_id(0);
+  // Local col ID (max: RTSN=TSN/WPTN).
+  const int_tp tidm = get_local_id(1);
+  // Work-group offset.
+  const int_tp offN = TSN * get_group_id(0);
+  // Work-group offset.
+  const int_tp offM = TSM * get_group_id(1);
+  // Local tile memory.
+  // Asub for loading weights & shuffling the output.
+  volatile __local Dtype Asub[TSM][TSK + v_pad_A];
```

Comment:
I wonder of Asub and BSub should be arrays of "int" rather than "char" for performance reasons. But this is probably something to be benchmarked later.